nping does not call anything nl_* related so it should not link against libnl (bug #529244)
Patching configure.ac would be best but eautoreconf does not handle the subdirs very well.

--- a/nping/configure.ac
+++ b/nping/configure.ac
@@ -114,9 +114,6 @@
 # OpenSSL and NSE C modules can require dlopen
 AC_SEARCH_LIBS(dlopen, dl)
 
-# libpcap can require libnl
-AC_SEARCH_LIBS(nl_handle_alloc, nl)
-
 # We test whether they specified openssl desires explicitly
 use_openssl="yes"
 specialssldir=""
